### PR TITLE
chore: fix percy types exports for newer typescript

### PR DESCRIPTION
**Pull request by Techn1x**
Fixes https://github.com/percy/cli/issues/1290

I have tested this via my own application, it resolves the error I was seeing detailed in the issue. I'm no expert at this but this makes sense to me.

Please let me know if there are other steps needed to have this released & published

### CODE REVIEW

File: packages/core/package.json [hunk 1]
```diff
@@ -24,7 +24,10 @@
   "types": "./types/index.d.ts",
```
Comment (Techn1x) [inline packages/core/package.json:24]: the original "types" remains here for backwards compatibility reasons with older TS

**Comment by itsjwala** [comment]
@Techn1x please make the changes in all the packages so it's uniform.

**Comment by Techn1x** [comment]
@itsjwala which package are you referring to specifically? I already applied the change to all packages in this monorepo that have types. I checked again just now in case I missed one.

That test that failed seems unrelated? Flaky?

**Review by itsjwala — APPROVED** [review]
@Techn1x 

* Apologies I didn't see the PR fully - config and core packages only contain types so that should do
*  yup windows tests are flakey, it should be fine to merge without it being green.

thanks for your contribution 🎉

**Pull request merged**